### PR TITLE
DAOS-11735 test: CR Pass 2 - Dangling rank entry test

### PR DESCRIPTION
**Pull request by shimizukko**
Test steps:

1. Create a pool and a container.
2. Write some data with IOR using SX.
3. Stop servers.
4. Remove pool directory from one of the mount points.
5. Enable checker.
6. Start checker.
7. Query the checker until expected number of inconsistencies are repaired.
8. Disable checker and start servers.
9. Wait for rebuild to finish.
10. Query the pool and verify that expected number of targets are disabled.

Add wait_for_check_complete() to refactor.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pool_membership
Test-repeat: 6
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/recovery/pool_membership.py [hunk 1]
```diff
@@ -295,3 +303,96 @@ def test_dangling_pool_map(self):
...
+        :avocado: tags=all,pr
+        :avocado: tags=hw,medium
+        :avocado: tags=recovery,pool_membership
+        :avocado: tags=PoolMembershipTest,test_dangling_rank_entry
+        """
+        targets = self.params.get("targets", "/run/server_config/engines/0/*")
+        exp_msg = "dangling rank entry"
+
+        # 1. Create a pool and a container.
+        # self.log_step("Create a pool and a container.")
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/pool_membership.py:332]: Uncommented? I recommend using `log_step` throughout this function

File: src/tests/ftest/recovery/pool_membership.py [hunk 2]
```diff
@@ -295,3 +303,96 @@ def test_dangling_pool_map(self):
...
+        self.pool = self.get_pool(connect=False)
+        self.container = self.get_container(pool=self.pool)
+
+        # 2. Write some data with IOR using SX.
+        self.ior_cmd.set_daos_params(self.server_group, self.pool, self.container.uuid)
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/pool_membership.py:337]: We should use labels unless there is a specific need
```suggestion
        self.ior_cmd.set_daos_params(self.server_group, self.pool, self.container.identifier)
```

File: src/tests/ftest/recovery/pool_membership.yaml [hunk 3]
```diff
@@ -12,12 +12,29 @@ server_config:
...
+ior:
+  client_processes:
+    ppn: 1
+  iorflags:
+    flags: -k -D 10 -v -w -W
+    api: DFS
+    transfer_size: 1M
+    block_size: 10G
+    dfs_oclass: SX
+    dfs_dir_oclass: SX
```
Comment (daltonbohning) [inline src/tests/ftest/recovery/pool_membership.yaml:40]: No need for `iorflags`
```suggestion
  flags: -k -D 10 -v -w -W
  api: DFS
  transfer_size: 1M
  block_size: 10G
  dfs_oclass: SX
  dfs_dir_oclass: SX
```

**Comment by daltonbohning** [comment]
@shimizukko Since this is a new test, and on a feature branch, do you mind addressing my comments?

File: src/tests/ftest/recovery/pool_membership.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/recovery/pool_membership.py]: Done

File: src/tests/ftest/recovery/pool_membership.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/recovery/pool_membership.py]: Done. I copied and pasted and forgot to update it.

File: src/tests/ftest/recovery/pool_membership.yaml (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/recovery/pool_membership.yaml]: Done. Thanks.

**Comment by shimizukko** [comment]
@daltonbohning Sorry I was busy with other work. I pushed a new commit to address your comments.

**Comment by daltonbohning** [comment]
> @daltonbohning Sorry I was busy with other work. I pushed a new commit to address your comments.

No worries. Thanks :)

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Pull request merged**